### ISSUE TITLE
Script error when assigning GOslim terms to DML-Gene overlaps

### ISSUE THREAD

**Issue by yaaminiv** [RobertsLab/resources#776]
I am trying to annotate DML-gene overlaps with GOslim terms for my *C. gigas* data in [this notebook](https://github.com/RobertsLab/project-gigas-oa-meth/blob/master/notebooks/2019-09-15-DML-Analysis.ipynb). I am unable to use `sort -V` to sort my blast output:

<img width="603" alt="Screen Shot 2019-10-29 at 11 08 15 AM" src="https://user-images.githubusercontent.com/22335838/67795973-6ae6c080-fa3c-11e9-8ad6-09fe2d007900.png">

I used this command previously in [this notebook](https://github.com/fish546-2018/yaamini-virginica/blob/master/notebooks/2019-08-28-blastx-to-GOslim.ipynb) for *C. virginica* data. Is there a different syntax I should use instead?

**Comment by kubu4** [comment]
This almost certainly related to using `sort` on a Macintosh. Are you running this on a different computer than you previously used? If yes, it's likely that the other Macintosh was actually using GNU `sort` that was installed using Homebrew. After that, the GNU `sort` was loaded in the system `$PATH` ahead of the default `sort` on the Macintosh. So, the end user doesn't realize that running `sort` is actually running GNU `sort`, instead of the `sort` version that's installed on the Macintosh.

TL;DR - Install GNU sort via Homebrew and see if the issue is resolved.

**Comment by yaaminiv** [comment]
@kubu4 I'm running this on the same computer I used for the *C. virginica* analyses (genefish). I installed Homebrew, shut down Jupyter notebook, and reopened my notebook and I still get the same error.

**Comment by kubu4** [comment]
After installing Homebrew, did you also install GNU sort?

Also, you might need to logout of whichever user profile you're logged in as and then log back in in order for the system $PATH to be updated.

**Comment by kubu4** [comment]
BTW, installing GNU sort is part of the `coreutils` "brew". Now that you have Homebrew installed, run this:

`brew install coreutils`

This will install a bunch of GNU utilities. To run GNU sort, you'll need to run the sort command as:

`gsort`

**Comment by kubu4** [comment]
> I'm running this on the same computer

Also, I should've clarified, the Homebrew installations are user-specific (I was assuming you were still using the same User profile). I see in your most recent notebook that you're logged in as a different User than you had previously used. So, you'll definitely need to install Homebrew and then use Hombrew to install the `coreutils` brew.

**Comment by yaaminiv** [comment]
I did not install GNU sort :laughing:. Went ahead and installed `coreutils` and now I'm logging out to retry it.

>  I see in your most recent notebook that you're logged in as a different User than you had previously used.

I'm still logged in as the same person! That's odd...

**Comment by kubu4** [comment]
![20191029_001](https://user-images.githubusercontent.com/4514104/67801310-252ef580-fa46-11e9-8236-9a0a59db91d9.png)

**Comment by yaaminiv** [comment]
Huh. I guess was working on my own computer (yaaminivenkataraman) which I know has Homebrew installed.

I logged back into my account on `genefish` and reran the code. Still get the error.

**Comment by kubu4** [comment]
What code did you run?

**Comment by yaaminiv** [comment]
I ran the code below since that's what worked in the previous notebook: 

`!awk '{print $3"\t"$2}' _blast-GO-unfolded.tab | sort -V > _blast-GO-unfolded.sorted`

I changed `sort` to `gsort` and it worked!:

`!awk '{print $3"\t"$2}' _blast-GO-unfolded.tab | gsort -V > _blast-GO-unfolded.sorted`

![Screen Shot 2019-10-29 at 12 37 00 PM](https://user-images.githubusercontent.com/22335838/67802645-cf0f8180-fa48-11e9-97d2-6b0d8445baa3.png)

**Comment by kubu4** [comment]
Like I [indicated above](https://github.com/RobertsLab/resources/issues/776#issuecomment-547584414), you need to run `gsort`. 

Good work! Glad you got it straightened out.
